eng/devices/ios.cake

```diff
@@ -430,9 +446,24 @@ void InstallIpa(string testApp, string testAppPackageName, string testDevice, st
 			var simulatorName = "XHarness";
 			Information("Looking for simulator: {0} iosversion {1}", simulatorName, iosVersionToRun);
 			var sims = ListAppleSimulators();
+
 			var simXH = sims.Where(s => s.Name.Contains(simulatorName) && s.Name.Contains(iosVersionToRun)).FirstOrDefault();
 			if (simXH == null)
-				throw new Exception("No simulator was found to run tests on.");
+			{
+				// if the device is already installed on this system then xharness won't create a new one
```

Comment:
This code is a little bit awkward. 

I found that if you already have a device that's "iPhone Xs 17.2" installed. Xharness won't create a new device and then this code just fails because it can't find something called "Xharness" 

Is there a way to get the UDID from the xharness script above?

@mandel-macaque ?